### ISSUE TITLE
pointer types error

### ISSUE THREAD

**Issue by Yujif1Aero** [lwfinger/rtl8852au#19]
Hi,
Thank you for sharing your great codes.
I tried to make your codes in CentOS8. However, I got error during make.

make ARCH=x86_64 CROSS_COMPILE= -C /lib/modules/4.18.0-408.el8.x86_64/build M=/home/username/rtl8852au  modules
make[1]: ディレクトリ '/usr/src/kernels/4.18.0-408.el8.x86_64' に入ります
  CC [M]  /home/username/rtl8852au/os_dep/linux/os_intfs.o
/home/username/rtl8852au/os_dep/linux/os_intfs.c:316:22: エラー: initialization of ‘u16 (*)(struct net_device *, struct sk_buff *, struct net_device *, u16 (*)(struct net_device *, struct sk_buff *, struct net_device *))’ {aka ‘short unsigned int (*)(struct net_device *, struct sk_buff *, struct net_device *, short unsigned int (*)(struct net_device *, struct sk_buff *, struct net_device *))’} from incompatible pointer type ‘u16 (*)(struct net_device *, struct sk_buff *, void *, u16 (*)(struct net_device *, struct sk_buff *, struct net_device *))’ {aka ‘short unsigned int (*)(struct net_device *, struct sk_buff *, void *, short unsigned int (*)(struct net_device *, struct sk_buff *, struct net_device *))’} [-Werror=incompatible-pointer-types]
  .ndo_select_queue = rtw_select_queue,
                      ^~~~~~~~~~~~~~~~
/home/username/rtl8852au/os_dep/linux/os_intfs.c:316:22: 備考: (‘rtw_netdev_ops.ndo_select_queue’ 用の初期化付近)
cc1: some warnings being treated as errors
make[2]: *** [scripts/Makefile.build:316: /home/username/rtl8852au/os_dep/linux/os_intfs.o] エラー 1
make[1]: *** [Makefile:1577: _module_/home/username/rtl8852au] エラー 2
make[1]: ディレクトリ '/usr/src/kernels/4.18.0-408.el8.x86_64' から出ます
make: *** [Makefile:639: modules] エラー 2

maybe, its error is from os_intfs.c. Could you fix it? Sorry, some words are in Japanese in error message .

Best,

**Comment by lwfinger** [comment]
No, kernel 4.18 is too old to backport this driver. You need to update to a newer kernel.

**Comment by Yujif1Aero** [comment]
> No, kernel 4.18 is too old to backport this driver. You need to update to a newer kernel.

Thank you for answering. It worked well since i got a newer kernel. By the way, I got any error due to #ifdef (__little endian). I fixed it to change #ifdef (__little endian) to #ifdef __little endian. Thank you so much.
